src/engine/src/scheduler/IScheduler.hpp

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include <entt/entt.hpp>
+
+namespace ES::Engine {
+class Registry;
+}
+
+namespace ES::Engine::Scheduler {
+using USystem = std::function<void(Registry &)>;
```

Comment:
Maybe put this declaration in a utils file (as you want, but I think it will be used at many location like Registry, all schedulers)

Reply:
would be great

Reply:
I tried and this is bad because of the same dependency problem, so I just removed the declaration from other schedulers. (they will get it from the IScheduler header anyway)